spec/requests/purchases/create_spec.rb

```diff
@@ -0,0 +1,141 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'POST /purchases/create', type: :request do
+  subject { response }
+
+  let!(:user) do
+    User.create!(
+      first_name: 'jane',
+      last_name: 'doe',
+      email: 'bar@baz.com',
+      password: '123456',
+      cpf: '1234567890',
+      iugu_id: '1',
+      address: 'foo',
+      address_number: '42',
+      zip_code: '123',
+      city: 'foo',
+      state: 'foo'
+    )
+  end
+
+  let(:course) do
+    Course.create!(
+      name: 'baz course',
+      description: 'some course description',
+      lesson_detail: 'foo',
+      subsection: subsection,
+      vimeo_id: '1234',
+      image_url: 'http://foo',
+      instructor: 'foo',
+      credentials: 'foo',
+      price: '1000',
+      discount_price: '900'
+    )
+  end
+
+  let!(:section) do
+    Section.create!(
+      name: 'foo section',
+      description: 'some section description',
+      image_url: 'http://image-url'
+    )
+  end
+
+  let!(:subsection) do
+    Subsection.create!(
+      name: 'bar subsection',
+      section: section
+    )
+  end
+
+  context 'when api charge fails' do
```

Comment:
Please add a `context 'without current user'` to make sure we're redirecting to sign up.